src/librustdoc/html/static/js/search.js

```diff
@@ -167,33 +504,142 @@ window.initSearch = function(rawSearchIndex) {
     }
 
     /**
-     * Executes the query and returns a list of results for each results tab.
-     * @param  {Object}        query          - The user query
-     * @param  {Array<string>} searchWords    - The list of search words to query against
-     * @param  {string}        [filterCrates] - Crate to search in
-     * @return {{
-     *   in_args: Array<?>,
-     *   returned: Array<?>,
-     *   others: Array<?>
-     * }}
+     * Parses the query.
+     *
+     * The supported syntax by this parser is as follow:
+     *
+     * ident = *(ALPHA / DIGIT / "_")
+     * path = ident *(DOUBLE-COLON ident)
+     * arg = path [generics]
+     * arg-without-generic = path
+     * type-sep = COMMA/WS *(COMMA/WS)
+     * nonempty-arg-list = arg *(type-sep arg) *(type-sep)
+     * nonempty-arg-list-without-generics = arg-without-generic *(type-sep arg-without-generic)
+     *                                      *(type-sep)
+     * generics = OPEN-ANGLE-BRACKET *WS [ nonempty-arg-list-without-generics ] *WS
```

Comment:
While this grammar says that the generics are required to be nonempty, the parser seems to accept an empty generics list:

```
$ node src/tools/rustdoc-js/tester.js         --doc-folder build/x86_64-unknown-linux-gnu/doc/         --crate-name std --dump 'a<>' --resource-suffix 1.61.0
{
    "original": "a<>",
    "userQuery": "a<>",
    "typeFilter": -1,
    "elems": [
        {
            "name": "a",
            "fullPath": [
                "a"
            ],
            "pathWithoutLast": [],
            "pathLast": "a",
            "generics": []
        }
    ],
    "returned": [],
    "foundElems": 1,
    "literalSearch": false,
    "error": null
}
```

Reply:
Then let's emit an error if generics are empty!

Reply:
Hum actually might be better to support it.

Reply:
Also, `[]` supposed to mean that it's optional, so I think it's as expected. Or did I miss something?

Reply:
You’re right. The square brackets mean optional. I misread the eBNF here.